addons/ui/fnc_openItemContextMenu.sqf

```diff
@@ -0,0 +1,192 @@
+    case "CFGGLASSES";
+    case "CFGWEAPONS": {
+        _options pushBack (GVAR(ItemContextMenuOptions) getVariable "#AllItems");
+    };
+};
+
+_options pushBack (GVAR(ItemContextMenuOptions) getVariable "#All");
+
+// Skip menu if no options.
+if (count _options isEqualTo 0) exitWith {};
+
+// ---
+// Create context menu.
+// ctrlSetBackgroundColor command does not seem to work for RscListBox.
+private _list = _display ctrlCreate [QGVAR(ItemContextMenu), -1];
+
+// ---
+// Populate context menu with options.
+{
+    _x params ["_slots", "_displayName", "_tooltip", "_color", "_icon", "_conditionEnable", "_conditionShow", "_statement", "_consume", "_params"];
+
+    private _args = [_unit, _container, _item, _slot, _params];
+    if (isLocalized _displayName) then {
+        _displayName = localize _displayName;
+    };
+
+    if (isLocalized _tooltip) then {
+        _tooltip = localize _tooltip;
+    };
+
+    if ((_slot in _slots || {"ALL" in _slots}) && {_args call _conditionShow}) then {
+        private _index = _list lbAdd _displayName;
+        _list lbSetTooltip [_index, _tooltip]; // broken, @todo
+
+        private _key = format [QGVAR(OptionData%1), _index];
+
+        if (_color isEqualTo []) then {
+            _color = getArray (configFile >> ctrlClassName _list >> "colorText");
+        };
+
+        _list lbSetColor [_index, _color];
+        _list lbSetData [_index, _key];
+        _list lbSetPicture [_index, _icon];
+
+        if !([_args call _conditionEnable] param [0, false]) then {
+            // Gray out.
+            _color = getArray (configFile >> ctrlClassName _list >> "colorDisabled");
+            _list lbSetColor [_index, _color];
+            _list lbSetSelectColor [_index, _color];
+
+            // Keep open, but do nothing. Also don't consume.
+            _conditionEnable = {true};
+            _statement = {true};
+            _consume = false;
+        };
+
+        _list setVariable [_key, [_conditionEnable, _statement, _consume, _args]];
+    };
+} forEach _options;
+
+// ---
+// Execute context menu option statement on selection.
+_list setVariable [QFUNC(activate), {
+    params ["_list", "_index"];
+    _list getVariable (_list lbData _index) params ["_condition", "_statement", "_consume", "_this"];
+
+    if (_this call _condition) then {
+        if (_consume) then {
+            params ["_unit", "_container", "_item", "_slot"];
+            switch _slot do {
+                case "UNIFORM_CONTAINER": {
+                    _unit removeItemFromUniform _item;
+                };
+                case "VEST_CONTAINER": {
+                    _unit removeItemFromVest _item;
+                };
+                case "BACKPACK_CONTAINER": {
+                    _unit removeItemFromBackpack _item;
+                };
+                default {
+                    ERROR_2("Cannot consume item %1 from %2.",_item,_slot);
+                    _statement = {true};
+                };
```

Comment:
Should be function that handles all kinds of slots.

Reply:
Inside, add a comment about considerations for remote objects (only relevant for CARGO and GROUND)